Source/Atomic/Core/Object.h

```diff
@@ -39,6 +39,29 @@ namespace Atomic
 class Context;
 class EventHandler;
 
+// ATOMIC BEGIN
+class Engine;

```

Comment:
I cringed a little at first introducing these forward decl in Object.h, though for the greater good, and we'll always know who the major players are... also, as purely passing pointers don't have to bring headers into Object.cpp, which means we're still pretty air tight :dash:

Very good...